libkbfs/bserver_remote.go

```diff
@@ -264,13 +264,16 @@ func (b *BlockServerRemote) Put(ctx context.Context, id BlockID, tlfID TlfID,
 		// BlockKey is misnamed -- it contains just the server
 		// half.
 		BlockKey: serverHalf.String(),
-		Folder:   tlfID.String(),
+		Folder:   md.ID.String(),
 		Buf:      buf,
 	}
 
 	err = b.client.PutBlock(ctx, arg)
 	if err != nil {
 		if qe, ok := err.(BServerErrorOverQuota); ok && !qe.Throttled {
+			name := md.GetTlfHandle().GetCanonicalName()
```

Comment:
hmm seems like such a shame to introduce a dependency on `RootMetadata` just to get the canonical name from it. For example, it's now more annoying to test, since one would have to build up an entire fake RMD to pass in. What about just passing in the canonical name or the TlfHandle instead?